src/librustc_mir/borrow_check/nll/universal_regions.rs

```diff
@@ -479,6 +482,14 @@ impl<'cx, 'gcx, 'tcx> UniversalRegionsBuilder<'cx, 'gcx, 'tcx> {
         let mut indices = self.compute_indices(fr_static, defining_ty);
         debug!("build: indices={:?}", indices);
 
+        let closure_base_def_id = self.infcx.tcx.closure_base_def_id(self.mir_def_id);
+
+        if self.mir_def_id != closure_base_def_id {
```

Comment:
This merits a comment. Something like

---

If this is a closure or generator, then the late-bound regions from the enclosing function are actually external regions to us. For example, here, `'a` is not local to the closure `c` (although it is local to the fn `foo`):

```
fn foo<'a>() {
    let c = || {
        let x: &'a u32 = ...;
    }
}
```